v4/core/base_service.go

```diff
@@ -180,11 +185,31 @@ func (service *BaseService) SetHTTPClient(client *http.Client) {
 }
 
 // DisableSSLVerification skips SSL verification.
+// This function sets a new http.Client instance on the service
+// and configures it to bypass verification of server certificates
+// and host names, making the client susceptible to "man-in-the-middle"
+// attacks.  This should be used only for testing.
 func (service *BaseService) DisableSSLVerification() {
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	client := DefaultHTTPClient()
+	tr, ok := client.Transport.(*http.Transport)
+	if tr != nil && ok {
+		tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 	}
-	service.Client.Transport = tr
+
+	service.SetHTTPClient(client)
+}
+
+// IsSSLDisabled returns true if and only if the service's http.Client instance
+// is configured to skip verification of server SSL certificates.
+func (service *BaseService) IsSSLDisabled() bool {
```

Comment:
While working on the corresponding generator changes (specifically the generated unit tests), I discovered that we need a more reliable way to check to see if a service has had ssl verification disabled or not, so I'm exposing this new public method on a BaseService.